docs/configuration.md

```diff
@@ -1966,67 +1078,54 @@ http:
-`/var/lib/registry` directory. Logging is in `debug` mode, which is the most
+This example configures the registry instance to run on port `5000`, binding to
+`localhost`, with the `debug` server enabled. Registry data is stored in the
+`/var/lib/registry` directory. Logging is set to `debug` mode, which is the most
 verbose.
 
-A similar simple configuration is available at
-[config-example.yml](https://github.com/docker/distribution/blob/master/cmd/registry/config-example.yml).
-Both are generally useful for local development.
+See
+[config-example.yml](https://github.com/docker/distribution/blob/master/cmd/registry/config-example.yml)
+for another simple configuration. Both examples are generally useful for local
+development.
 
 
 ## Example: Middleware configuration
 
-This example illustrates how to configure storage middleware in a registry.
-Middleware allows the registry to serve layers via a content delivery network
-(CDN). This is useful for reducing requests to the storage layer.
-
-The registry supports [Amazon
-Cloudfront](http://aws.amazon.com/cloudfront/). You can only use Cloudfront in
-conjunction with the S3 storage driver.
-
-<table>
-  <tr>
-    <th>Parameter</th>
-    <th>Description</th>
-  </tr>
-  <tr>
-    <td><code>name</code></td>
-    <td>The storage middleware name. Currently <code>cloudfront</code> is an accepted value.</td>
-  </tr>
-  <tr>
-    <td><code>disabled</code></td>
-    <td>Set to <code>false</code> to easily disable the middleware.</td>
-  </tr>
-  <tr>
-    <td><code>options:</code></td>
-    <td>
-    A set of key/value options to configure the middleware.
-    <ul>
-    <li><code>baseurl:</code> The Cloudfront base URL.</li>
-    <li><code>privatekey:</code> The location of your AWS private key on the filesystem. </li>
-    <li><code>keypairid:</code> The ID of your Cloudfront keypair. </li>
-    <li><code>duration:</code> The duration in minutes for which the URL is valid. Default is 20. </li>
-    </ul>
-    </td>
-  </tr>
-</table>
-
-The following example illustrates these values:
+This example configures storage middleware in a registry. Middleware allows the
+registry to serve layers via a content delivery network (CDN). This reduces
+requests to the storage layer.
 
-```
+The registry supports
+[Amazon Cloudfront](http://aws.amazon.com/cloudfront/). Cloudfront requires the
+S3 storage driver.
+
+This table describes the configuration:
+
+| Parameter | Description                                           |
+|-----------|-------------------------------------------------------|
+| `name`    | The name of the storage middleware. `cloudfront` is an accepted value. |
+| `disabled`| Set to `true` to easily disable the middleware.       |
+| `options` | A set of key-value options to configure the middleware. The rest of the rows of this table are the options. |
+| `baseurl` | The CloudFront base URL.                              |
```

Comment:
This section is still kind of confusing.  Since these are `cloudfront` specific options they should probably be broken out into a completely separate section for `cloudfront` options. We do have another PR open for another type of middleware so it is probably best to break this up.